src/mbgl/text/placement.cpp

```diff
@@ -595,53 +594,60 @@ void Placement::placeSymbolBucket(const BucketPlacementData& params, std::set<ui
 
-            Point<float> offset{};
-            if (variableAnchor) {
-                const CollisionBox& textBox = symbol.textCollisionFeature.boxes.front();
-                float width = textBox.x2 - textBox.x1;
-                float height = textBox.y2 - textBox.y1;
-                offset = calculateVariableLayoutOffset(*variableAnchor,
-                                                       width,
-                                                       height,
-                                                       symbol.variableTextOffset,
-                                                       symbol.textBoxScale,
-                                                       rotateTextWithMap,
-                                                       pitchTextWithMap,
-                                                       state.getBearing());
+            bool intersects = false;
+            if (!symbol.textCollisionFeature.boxes.empty()) {
+                const auto& textCollisionBox = symbol.textCollisionFeature.boxes.front();
+
+                Point<float> offset{};
+                if (variableAnchor) {
+                    float width = textCollisionBox.x2 - textCollisionBox.x1;
+                    float height = textCollisionBox.y2 - textCollisionBox.y1;
+                    offset = calculateVariableLayoutOffset(*variableAnchor,
+                                                           width,
+                                                           height,
+                                                           symbol.variableTextOffset,
+                                                           symbol.textBoxScale,
+                                                           rotateTextWithMap,
+                                                           pitchTextWithMap,
+                                                           state.getBearing());
+                }
+                intersects = collisionBoxIntersectsTileEdges(textCollisionBox, offset);
             }
 
-            bool intercects =
-                collisionIndex.intercectsTileEdges(collisionBox, offset, renderTile.matrix, pixelRatio, *tileBorders);
-            if (!intercects) {
-                // Check if this symbol intersects the neighbor tile borders.
-                // If so, it also shall be placed with priority (location = FeatureLocation::IntersectsTileBorder).
-                for (const auto& neighbor : neightbors) {
-                    intercects = collisionIndex.intercectsTileEdges(
-                        collisionBox, offset + neighbor.shift, neighbor.matrix, pixelRatio, neighbor.borders);
-                    if (intercects) break;
-                }
+            if (!symbol.iconCollisionFeature.boxes.empty() && !intersects) {
+                const auto& iconCollisionBox = symbol.iconCollisionFeature.boxes.front();
+                intersects = collisionBoxIntersectsTileEdges(iconCollisionBox, {});
```

Comment:
In case if variable placement + icon-text-fit is used, icon collistion box has to be shifted as well. Check `doVariableIconPlacement` [branch](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/text/placement.cpp#L380). Maybe worth adding render test for that case?

Reply:
variable placement + icon-text-fit is not yet supported in tile mode :disappointed: , however I'd better track this effort in a separate PR, would it lgtu?

Reply:
Sure, I think it should be fine to split a fix into a separate PRs to avoid huge diff / regressions.